base/deprecated.jl

```diff
@@ -1788,7 +1788,53 @@ end
     @deprecate get_creds!(cache::CachedCredentials, credid, default) get!(cache, credid, default)
 end
 
-@deprecate eye(::Type{Diagonal{T}}, n::Int) where {T} Diagonal{T}(I, n)
+## goodbeye, eye!
+export eye
```

Comment:
:+1:

Reply:
In theory, I guess we also need
```
@eval Base.LinAlg import Base.eye
@eval Base.SparseArrays import Base.eye
```
if someone uses `LinAlg.eye(...)` or `SparseArrays.eye()` in their codes. I don't think we have done that before, but perhaps we should, since changes like this should not break code :)